meta-lmp-base/recipes-core/initrdscripts/initramfs-module-install-efi/init-install-efi.sh

```diff
@@ -138,7 +138,12 @@ if [ ! -e /etc/mtab ] && [ -e /proc/mounts ]; then
     ln -sf /proc/mounts /etc/mtab
 fi
 
-disk_size=$(parted -s ${device} unit mb print | grep '^Disk .*: .*MB' | cut -d" " -f 3 | sed -e "s/MB//")
+disk_info=$(parted -s ${device} unit mb print 2>&1 || true)
+if echo ${disk_info} | grep -q "unrecognised disk label"; then
+	echo "No valid disk label found, creating one"
+	parted -s ${device} -- mklabel gpt
+fi
+disk_size=$(echo ${disk_info} | grep '^Disk .*: .*MB' | cut -d" " -f 3 | sed -e "s/MB//" || true)
```

Comment:
I would remove this last `|| true`. Let the script still fail if there is some error in the grep/cut/sed sequence, since we do need to have disk_size set

Reply:
Yes, this is a leftover, we only need that on the first call, will remove.